utils/rtl-text-detection.ts

```diff
@@ -0,0 +1,18 @@
+const rtlChars = [
+  /* arabic ranges */
+  '\u0600-\u06FF',
+  '\u0750-\u077F',
+  '\uFB50-\uFDFF',
+  '\uFE70-\uFEFF',
+  /* hebrew range */
+  '\u05D0-\u05FF',
+].join('')
+
+const reRTL = new RegExp(`[${rtlChars}]`, 'g')
+
+export function isRtlText(text: string) {
+  const textCount = text.replace(/[0-9\s\\\/.,\-+="']/g, '').length
```

Comment:
Can we make this more performant by first checking if there are any RTL characters and early returning if not?

Reply:
done